src/main/java/org/spongepowered/api/util/Color.java

```diff
@@ -191,39 +216,64 @@ public static Color mixColors(Color... colors) {
         return ofRgb(averageRed, averageGreen, averageBlue);
     }
 
-    private final byte red;
-    private final byte green;
-    private final byte blue;
-    private final int rgb;
-
-    private Color(int red, int green, int blue) {
-        this.red = (byte) (red & MASK);
-        this.green = (byte) (green & MASK);
-        this.blue = (byte) (blue & MASK);
-        this.rgb = ((this.red & MASK) << 0x10)
-                   | ((this.green & MASK) << 0x8)
-                   | ((this.blue & MASK) << 0);
+    /**
+     * Creates a new {@link Color} combining the provided {@link DyeColor}
+     * objects. Since {@link DyeColor}s can be converted into {@link Color}
+     * objects themselves, their summation and average is taken into effect
+     * to properly mix the colors together.
+     *
+     * @param colors The colors to mix
+     * @return The final output mixed color
+     */
+    public static Color mixDyeColors(DyeColor... colors) {
+        Validate.noNullElements(colors, "No nulls allowed!");
+        final Color[] actualColors = new Color[colors.length];
+        for (int i = 0; i < colors.length; i++) {
+            actualColors[i] = colors[i].getColor();
+        }
+
+        return mixColors(actualColors);
+    }
+
+    private final int hex;
+    private final int red;
+    private final int green;
+    private final int blue;
+
+    private Color(int hex, int red, int green, int blue) {
+        this.hex = hex;
+        this.red = red;
+        this.green = green;
+        this.blue = blue;
     }
 
     /**
-     * Gets the {@code red} value of this {@link Color}.
+     * Gets the {@code hex} value of this {@link Color}.
      *
-     * @return The red value
+     * @return The color in hexadecimal
      */
-    public int getRed() {
-        return MASK & this.red;
+    public int getHex() {
```

Comment:
I'm not sure why we're renaming this as `getRGB` is a lot more descriptive than `getHex`. Especially since the javadocs don't even mention the ordering or what the values are (ARGB, ABGR, RGB, BGR, HSV, CMYK...),

Reply:
I felt that `getHex` more accurately described what was being returned at first, but having a better understanding of color theory now I agree with this.